src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/ExplosionsListener.java

```diff
@@ -50,22 +53,44 @@ public void onBlockExplode(BlockExplodeEvent e) {
         removeResistantBlocks(e.blockList().iterator());
     }
 
+    @EventHandler(priority = EventPriority.HIGHEST, ignoreCancelled = true)
+    public void onEntityBreak(EntityChangeBlockEvent e) {
+        if (e.getEntity().getType() == EntityType.WITHER || e.getEntity().getType() == EntityType.WITHER_SKULL) {
+
+            removeResistantBlock(e.getBlock());
+        }
+    }
+
     private void removeResistantBlocks(@Nonnull Iterator<Block> blocks) {
         while (blocks.hasNext()) {
             Block block = blocks.next();
             SlimefunItem item = BlockStorage.check(block);
 
             if (item != null) {
                 blocks.remove();
-
-                if (!(item instanceof WitherProof) && !item.callItemHandler(BlockBreakHandler.class, handler -> handleExplosion(handler, block))) {
-                    BlockStorage.clearBlockInfo(block);
-                    block.setType(Material.AIR);
-                }
+                removeResistantBlock(block);
```

Comment:
The BlockStorage check is still being made twice, you can just pass the `SlimefunItem` instance here.